docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,277 @@
+# Firecracker Snapshotting
+
+## What is microVM snapshotting?
+
+MicroVM snapshotting is a mechanism through which a running microVM, including
+the guest environment, can be serialized and saved to an external medium in the
+form of a `snapshot`. This snapshot can be later used to restore a microVM
+with its guest workload at that particular point in time.
+
+## Snapshotting in Firecracker
+
+Firecracker snapshotting allows the creation of a microVM snapshot. This
+snapshot can be later loaded in a possibly different Firecracker process and
+the original guest workload is simply resumed. It is not guaranteed though that
+the state of the network connections survives the process.
+
+In order to make restoring possible, Firecracker snapshots save the full state
+of the following resources:
+- the guest memory,
+- the emulated HW state (both KVM and Firecracker emulated HW),
+- the guest's storage disks' contents.
+
+The state of the components listed above is generated independently, which brings
+flexibility to our snapshotting support. This means that taking a snapshot results
+in multiple files that are composing the full microVM snapshot:
+- the guest memory file,
+- the microVM state file,
+- zero or more disk files (depending on how many the guest had; these are
+  **managed by the users**, which means they need to externally back up their
+  block devices backing files).
+
+This design allows sharing any read-only resources (such as read-only disks or
+microVM state) between multiple microVMs in order to improve density.
+When loading a snapshot, instead of loading at resume time the full contents from
+file to memory, Firecracker creates a `MAP_PRIVATE` mapping of the memory file,
```

Comment:
A link to `mmap` reference would be useful here, as supporting explanation for what `MAP_PRIVATE` means